### ISSUE TITLE
[Bug?]: No type hints helping me when I add a field to my SDL, but forget to include it in the GQL resolver

### ISSUE THREAD

**Issue by Tobbe** [redwoodjs/redwood#5982]
### What's not working?

The SDL is not used as the source of truth for what the GQL api should look like when it comes to TS types (see the reproduction steps below for a clear example of what's amiss)

@dac09 wrote a comment related to this here https://github.com/redwoodjs/redwood/pull/5491#issuecomment-1176449311
To save you the click, I've pasted it below (only edited for spelling)

---

@Tobbe based on the issue you brought up the other day I think I've identified a shortcoming in the resolverFn, we do need to keep it as a Partial unfortunately, because the mapped prisma types don't take into account which of the fields in the SDL are optional.

But... I do have a suggestion for your particular case. If you are modifying a mapped type i.e. An SDL model that also exists in Prisma, but you're adding additional fields to your SDL model - you can just define a different type, for example:

```gql
  type Post {
    id: Int!
    title: String!
    body: String!

  }

  type PostWithAdditionalFields {
    id: Int!
    title: String!
    body: String!
    bazinga: Boolean # <-- new field
  }


  type Mutation {
    updatePost(id: Int!, input: UpdatePostInput!): PostWithAdditionalFields! @requireAuth
```
    
This will still generate the MutationResolvers['updatePost'] with your new type, but of course if you use DateTimes, that's not going to be so easy to use.

### How do we reproduce the bug?

1. Create a new RW app `yarn create redwood-app --ts sdl-fields`

2. Add a `User` model to `schema.prisma`
   ```prisma
   model User {
     id       Int      @id @default(autoincrement())
     email    String   @unique
     name     String?
     birthday DateTime
   }
   ```

3. Create a migration `yarn rw prisma migrate dev -n user`

4. Run the sdl generator `yarn rw g sdl User`

5. Now update `users.sdl.ts` and add an `age` field to the user
   ```gql
   type User {
     id: Int!
     email: String!
     name: String
     birthday: DateTime!
     age: Int!
   }
   ```

6. I should now get type errors in my GQL resolvers, because they no longer return what the SDL says they should return.

### Additional info

One might argue that the correct type would be one that's generated based on the SDL. And we actually do generate types based on the SDL too.

It looks like this

```ts
export type User = {
  __typename?: 'User';
  age: Scalars['Int'];
  birthday: Scalars['DateTime'];
  email: Scalars['String'];
  id: Scalars['Int'];
  name?: Maybe<Scalars['String']>;
};
```

So let's update our resolver to use that type instead:
```ts
export const user = async ({ id }): Promise<User> => {
  const user = await db.user.findUnique({
    where: { id },
  })

  return user
}
```

Doing that we get red squiggles, which is exactly what we want!
![image](https://user-images.githubusercontent.com/30793/179394981-8c2454fa-8921-41b9-a450-bef2a47c4a45.png)
And also a helpful error message
```
Property 'age' is missing in type 'import("/Users/tobbe/tmp/sdl-fields/node_modules/.prisma/client/index").User' but required in type 'import("/Users/tobbe/tmp/sdl-fields/api/types/graphql").User'.ts(2741)
graphql.d.ts(85, 3): 'age' is declared here.
```

This is looking great so far! Now let's try to fix the error by adding a hard-coded age of 15.

![image](https://user-images.githubusercontent.com/30793/179395059-aa9f71a9-9468-4ed3-9e6a-0cdda18fae68.png)

Ohh, no! It's still red 🙁 What's wrong?

```
Type '{ age: number; id: number; email: string; name: string; birthday: Date; }' is not assignable to type 'User'.
  Types of property 'birthday' are incompatible.
    Type 'Date' is not assignable to type 'string'.ts(2322)
```

And this is exactly what the generated `QueryResolvers['...']` type does for us. It helps us handle the `Date` <-> `string` incompatibility. (`QueryResolvers` also helps us type the input parameters. And not just the `{ id }` you see in the example I've been using here, but also the extra parameters you can pass to gql resolvers, like `root`, `context` and `info`.)

### What's your environment? (If it applies)

```shell
System:
  OS: macOS 12.4
  Shell: 5.8.1 - /bin/zsh
Binaries:
  Node: 16.14.2 - /private/var/folders/w7/hkfhzp4x68qcbr1ggchr7hx40000gn/T/xfs-0e50f8ec/node
  Yarn: 3.2.1 - /private/var/folders/w7/hkfhzp4x68qcbr1ggchr7hx40000gn/T/xfs-0e50f8ec/yarn
Databases:
  SQLite: 3.37.0 - /usr/bin/sqlite3
Browsers:
  Chrome: 103.0.5060.114
  Edge: 103.0.1264.62
  Firefox: 99.0.1
  Safari: 15.5
npmPackages:
  @redwoodjs/core: 2.0.0 => 2.0.0
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by Tobbe** [comment]
@dac09 I assigned you for triage. Doesn't have to be you who fix this (if we indeed should change anything at all). And this isn't super high prio for me. So no need to drop anything else anyone is working on just to fix this right now

**Comment by dac09** [comment]
Closing as this is resolved by #6228
